// facehash/index.js
import { t as stringHash } from "./hash.js";
import * as React from "react";
import { jsx, jsxs } from "react/jsx-runtime";

//#region src/faces.tsx
const BLINK_KEYFRAMES = `
@keyframes facehash-blink {
	0%, 92%, 100% { transform: scaleY(1); }
	96% { transform: scaleY(0.05); }
}
`;
let keyframesInjected = false;
function injectBlinkKeyframes() {
	if (keyframesInjected || typeof document === "undefined") return;
	const style = document.createElement("style");
	style.textContent = BLINK_KEYFRAMES;
	document.head.appendChild(style);
	keyframesInjected = true;
}
function getBlinkStyle(timing) {
	if (!timing) return {};
	return {
		animation: `facehash-blink ${timing.duration}s ease-in-out ${timing.delay}s infinite`,
		transformOrigin: "center center"
	};
}
/**
* Round eyes face - simple circular eyes
*/
const RoundFace = ({ className, style, enableBlink, blinkTimings }) => {
	if (enableBlink) injectBlinkKeyframes();
	return /* @__PURE__ */ jsxs("svg", {
		"aria-hidden": "true",
		className,
		fill: "none",
		style,
		viewBox: "0 0 63 15",
		xmlns: "http://www.w3.org/2000/svg",
		children: [
			/* @__PURE__ */ jsx("title", { children: "Round Eyes" }),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.right) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M62.4 7.2C62.4 11.1765 59.1765 14.4 55.2 14.4C51.2236 14.4 48 11.1765 48 7.2C48 3.22355 51.2236 0 55.2 0C59.1765 0 62.4 3.22355 62.4 7.2Z",
					fill: "currentColor"
				})
			}),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.left) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M14.4 7.2C14.4 11.1765 11.1765 14.4 7.2 14.4C3.22355 14.4 0 11.1765 0 7.2C0 3.22355 3.22355 0 7.2 0C11.1765 0 14.4 3.22355 14.4 7.2Z",
					fill: "currentColor"
				})
			})
		]
	});
};
/**
* Cross eyes face - X-shaped eyes
*/
const CrossFace = ({ className, style, enableBlink, blinkTimings }) => {
	if (enableBlink) injectBlinkKeyframes();
	return /* @__PURE__ */ jsxs("svg", {
		"aria-hidden": "true",
		className,
		fill: "none",
		style,
		viewBox: "0 0 71 23",
		xmlns: "http://www.w3.org/2000/svg",
		children: [
			/* @__PURE__ */ jsx("title", { children: "Cross Eyes" }),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.left) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M11.5 0C12.9411 0 13.6619 0.000460386 14.1748 0.354492C14.3742 0.49213 14.547 0.664882 14.6846 0.864258C15.0384 1.37711 15.0391 2.09739 15.0391 3.53809V7.96094H19.4619C20.9027 7.96094 21.6229 7.9615 22.1357 8.31543C22.3352 8.45308 22.5079 8.62578 22.6455 8.8252C22.9995 9.3381 23 10.0589 23 11.5C23 12.9408 22.9995 13.661 22.6455 14.1738C22.5079 14.3733 22.3352 14.5459 22.1357 14.6836C21.6229 15.0375 20.9027 15.0381 19.4619 15.0381H15.0391V19.4619C15.0391 20.9026 15.0384 21.6229 14.6846 22.1357C14.547 22.3351 14.3742 22.5079 14.1748 22.6455C13.6619 22.9995 12.9411 23 11.5 23C10.0592 23 9.33903 22.9994 8.82617 22.6455C8.62674 22.5079 8.45309 22.3352 8.31543 22.1357C7.96175 21.6229 7.96191 20.9024 7.96191 19.4619V15.0381H3.53809C2.0973 15.0381 1.37711 15.0375 0.864258 14.6836C0.664834 14.5459 0.492147 14.3733 0.354492 14.1738C0.000498831 13.661 -5.88036e-08 12.9408 0 11.5C6.2999e-08 10.0589 0.000460356 9.3381 0.354492 8.8252C0.492144 8.62578 0.664842 8.45308 0.864258 8.31543C1.37711 7.9615 2.09731 7.96094 3.53809 7.96094H7.96191V3.53809C7.96191 2.09765 7.96175 1.37709 8.31543 0.864258C8.45309 0.664828 8.62674 0.492149 8.82617 0.354492C9.33903 0.000555366 10.0592 1.62347e-09 11.5 0Z",
					fill: "currentColor"
				})
			}),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.right) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M58.7695 0C60.2107 0 60.9314 0.000460386 61.4443 0.354492C61.6437 0.49213 61.8165 0.664882 61.9541 0.864258C62.308 1.37711 62.3086 2.09739 62.3086 3.53809V7.96094H66.7314C68.1722 7.96094 68.8924 7.9615 69.4053 8.31543C69.6047 8.45308 69.7774 8.62578 69.915 8.8252C70.2691 9.3381 70.2695 10.0589 70.2695 11.5C70.2695 12.9408 70.269 13.661 69.915 14.1738C69.7774 14.3733 69.6047 14.5459 69.4053 14.6836C68.8924 15.0375 68.1722 15.0381 66.7314 15.0381H62.3086V19.4619C62.3086 20.9026 62.308 21.6229 61.9541 22.1357C61.8165 22.3351 61.6437 22.5079 61.4443 22.6455C60.9314 22.9995 60.2107 23 58.7695 23C57.3287 23 56.6086 22.9994 56.0957 22.6455C55.8963 22.5079 55.7226 22.3352 55.585 22.1357C55.2313 21.6229 55.2314 20.9024 55.2314 19.4619V15.0381H50.8076C49.3668 15.0381 48.6466 15.0375 48.1338 14.6836C47.9344 14.5459 47.7617 14.3733 47.624 14.1738C47.27 13.661 47.2695 12.9408 47.2695 11.5C47.2695 10.0589 47.27 9.3381 47.624 8.8252C47.7617 8.62578 47.9344 8.45308 48.1338 8.31543C48.6466 7.9615 49.3668 7.96094 50.8076 7.96094H55.2314V3.53809C55.2314 2.09765 55.2313 1.37709 55.585 0.864258C55.7226 0.664828 55.8963 0.492149 56.0957 0.354492C56.6086 0.000555366 57.3287 1.62347e-09 58.7695 0Z",
					fill: "currentColor"
				})
			})
		]
	});
};
/**
* Line eyes face - horizontal line eyes
*/
const LineFace = ({ className, style, enableBlink, blinkTimings }) => {
	if (enableBlink) injectBlinkKeyframes();
	return /* @__PURE__ */ jsxs("svg", {
		"aria-hidden": "true",
		className,
		fill: "none",
		style,
		viewBox: "0 0 82 8",
		xmlns: "http://www.w3.org/2000/svg",
		children: [
			/* @__PURE__ */ jsx("title", { children: "Line Eyes" }),
			/* @__PURE__ */ jsxs("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.left) : void 0,
				children: [/* @__PURE__ */ jsx("path", {
					d: "M3.53125 0.164063C4.90133 0.164063 5.58673 0.163893 6.08301 0.485352C6.31917 0.638428 6.52075 0.840012 6.67383 1.07617C6.99555 1.57252 6.99512 2.25826 6.99512 3.62891C6.99512 4.99911 6.99536 5.68438 6.67383 6.18066C6.52075 6.41682 6.31917 6.61841 6.08301 6.77148C5.58672 7.09305 4.90147 7.09277 3.53125 7.09277C2.16062 7.09277 1.47486 7.09319 0.978516 6.77148C0.742356 6.61841 0.540772 6.41682 0.387695 6.18066C0.0662401 5.68439 0.0664063 4.999 0.0664063 3.62891C0.0664063 2.25838 0.0660571 1.57251 0.387695 1.07617C0.540772 0.840012 0.742356 0.638428 0.978516 0.485352C1.47485 0.163744 2.16076 0.164063 3.53125 0.164063Z",
					fill: "currentColor"
				}), /* @__PURE__ */ jsx("path", {
					d: "M25.1836 0.164063C26.5542 0.164063 27.24 0.163638 27.7363 0.485352C27.9724 0.638384 28.1731 0.8401 28.3262 1.07617C28.6479 1.57252 28.6484 2.25825 28.6484 3.62891C28.6484 4.99931 28.6478 5.68436 28.3262 6.18066C28.1731 6.41678 27.9724 6.61842 27.7363 6.77148C27.24 7.09321 26.5542 7.09277 25.1836 7.09277H11.3262C9.95557 7.09277 9.26978 7.09317 8.77344 6.77148C8.53728 6.61841 8.33569 6.41682 8.18262 6.18066C7.86115 5.68438 7.86133 4.99902 7.86133 3.62891C7.86133 2.25835 7.86096 1.57251 8.18262 1.07617C8.33569 0.840012 8.53728 0.638428 8.77344 0.485352C9.26977 0.163768 9.95572 0.164063 11.3262 0.164063H25.1836Z",
					fill: "currentColor"
				})]
			}),
			/* @__PURE__ */ jsxs("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.right) : void 0,
				children: [/* @__PURE__ */ jsx("path", {
					d: "M78.2034 7.09325C76.8333 7.09325 76.1479 7.09342 75.6516 6.77197C75.4155 6.61889 75.2139 6.4173 75.0608 6.18114C74.7391 5.6848 74.7395 4.99905 74.7395 3.62841C74.7395 2.2582 74.7393 1.57294 75.0608 1.07665C75.2139 0.840493 75.4155 0.638909 75.6516 0.485832C76.1479 0.164271 76.8332 0.164543 78.2034 0.164543C79.574 0.164543 80.2598 0.164122 80.7561 0.485832C80.9923 0.638909 81.1939 0.840493 81.347 1.07665C81.6684 1.57293 81.6682 2.25831 81.6682 3.62841C81.6682 4.99894 81.6686 5.68481 81.347 6.18114C81.1939 6.4173 80.9923 6.61889 80.7561 6.77197C80.2598 7.09357 79.5739 7.09325 78.2034 7.09325Z",
					fill: "currentColor"
				}), /* @__PURE__ */ jsx("path", {
					d: "M56.5511 7.09325C55.1804 7.09325 54.4947 7.09368 53.9983 6.77197C53.7622 6.61893 53.5615 6.41722 53.4085 6.18114C53.0868 5.6848 53.0862 4.99907 53.0862 3.62841C53.0862 2.258 53.0868 1.57296 53.4085 1.07665C53.5615 0.840539 53.7622 0.638898 53.9983 0.485832C54.4947 0.164105 55.1804 0.164543 56.5511 0.164543H70.4085C71.7791 0.164543 72.4649 0.164146 72.9612 0.485832C73.1974 0.638909 73.399 0.840493 73.552 1.07665C73.8735 1.57293 73.8733 2.25829 73.8733 3.62841C73.8733 4.99896 73.8737 5.68481 73.552 6.18114C73.399 6.4173 73.1974 6.61889 72.9612 6.77197C72.4649 7.09355 71.7789 7.09325 70.4085 7.09325H56.5511Z",
					fill: "currentColor"
				})]
			})
		]
	});
};
/**
* Curved eyes face - sleepy/happy curved eyes
*/
const CurvedFace = ({ className, style, enableBlink, blinkTimings }) => {
	if (enableBlink) injectBlinkKeyframes();
	return /* @__PURE__ */ jsxs("svg", {
		"aria-hidden": "true",
		className,
		fill: "none",
		style,
		viewBox: "0 0 63 9",
		xmlns: "http://www.w3.org/2000/svg",
		children: [
			/* @__PURE__ */ jsx("title", { children: "Curved Eyes" }),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.left) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M0 5.06511C0 4.94513 0 4.88513 0.00771184 4.79757C0.0483059 4.33665 0.341025 3.76395 0.690821 3.46107C0.757274 3.40353 0.783996 3.38422 0.837439 3.34559C2.40699 2.21129 6.03888 0 10.5 0C14.9611 0 18.593 2.21129 20.1626 3.34559C20.216 3.38422 20.2427 3.40353 20.3092 3.46107C20.659 3.76395 20.9517 4.33665 20.9923 4.79757C21 4.88513 21 4.94513 21 5.06511C21 6.01683 21 6.4927 20.9657 6.6754C20.7241 7.96423 19.8033 8.55941 18.5289 8.25054C18.3483 8.20676 17.8198 7.96876 16.7627 7.49275C14.975 6.68767 12.7805 6 10.5 6C8.21954 6 6.02504 6.68767 4.23727 7.49275C3.18025 7.96876 2.65174 8.20676 2.47108 8.25054C1.19668 8.55941 0.275917 7.96423 0.0342566 6.6754C0 6.4927 0 6.01683 0 5.06511Z",
					fill: "currentColor"
				})
			}),
			/* @__PURE__ */ jsx("g", {
				style: enableBlink ? getBlinkStyle(blinkTimings?.right) : void 0,
				children: /* @__PURE__ */ jsx("path", {
					d: "M42 5.06511C42 4.94513 42 4.88513 42.0077 4.79757C42.0483 4.33665 42.341 3.76395 42.6908 3.46107C42.7573 3.40353 42.784 3.38422 42.8374 3.34559C44.407 2.21129 48.0389 0 52.5 0C56.9611 0 60.593 2.21129 62.1626 3.34559C62.216 3.38422 62.2427 3.40353 62.3092 3.46107C62.659 3.76395 62.9517 4.33665 62.9923 4.79757C63 4.88513 63 4.94513 63 5.06511C63 6.01683 63 6.4927 62.9657 6.6754C62.7241 7.96423 61.8033 8.55941 60.5289 8.25054C60.3483 8.20676 59.8198 7.96876 58.7627 7.49275C56.975 6.68767 54.7805 6 52.5 6C50.2195 6 48.025 6.68767 46.2373 7.49275C45.1802 7.96876 44.6517 8.20676 44.4711 8.25054C43.1967 8.55941 42.2759 7.96423 42.0343 6.6754C42 6.4927 42 6.01683 42 5.06511Z",
					fill: "currentColor"
				})
			})
		]
	});
};
/**
* All available face components
*/
const FACES = [
	RoundFace,
	CrossFace,
	LineFace,
	CurvedFace
];

//#endregion
//#region src/facehash.tsx
const INTENSITY_PRESETS = {
	none: {
		rotateRange: 0,
		translateZ: 0,
		perspective: "none"
	},
	subtle: {
		rotateRange: 5,
		translateZ: 4,
		perspective: "800px"
	},
	medium: {
		rotateRange: 10,
		translateZ: 8,
		perspective: "500px"
	},
	dramatic: {
		rotateRange: 15,
		translateZ: 12,
		perspective: "300px"
	}
};
const SPHERE_POSITIONS = [
	{
		x: -1,
		y: 1
	},
	{
		x: 1,
		y: 1
	},
	{
		x: 1,
		y: 0
	},
	{
		x: 0,
		y: 1
	},
	{
		x: -1,
		y: 0
	},
	{
		x: 0,
		y: 0
	},
	{
		x: 0,
		y: -1
	},
	{
		x: -1,
		y: -1
	},
	{
		x: 1,
		y: -1
	}
];
const DEFAULT_GRADIENT_STYLE = { background: "radial-gradient(ellipse 100% 100% at 50% 50%, rgba(255,255,255,0.15) 0%, transparent 60%)" };
/**
* Facehash - Deterministic avatar faces from any string.
*
* @example
* ```tsx
* // With Tailwind classes
* <Facehash
*   name="John"
*   colorClasses={["bg-pink-500", "bg-blue-500"]}
* />
*
* // With hex colors
* <Facehash
*   name="John"
*   colors={["#ec4899", "#3b82f6"]}
* />
*
* // Plain color (no gradient)
* <Facehash name="John" variant="solid" />
* ```
*/
const Facehash = React.forwardRef(({ name, size = 40, variant = "gradient", intensity3d = "dramatic", interactive = true, showInitial = true, colors, colorClasses, gradientOverlayClass, onRenderMouth, enableBlink = false, className, style, onMouseEnter, onMouseLeave, ...props }, ref) => {
	const [isHovered, setIsHovered] = React.useState(false);
	const { FaceComponent, colorIndex, rotation, blinkTimings } = React.useMemo(() => {
		const hash = stringHash(name);
		const faceIndex = hash % FACES.length;
		const _colorIndex = hash % (colorClasses?.length ?? colors?.length ?? 1);
		const position = SPHERE_POSITIONS[hash % SPHERE_POSITIONS.length] ?? {
			x: 0,
			y: 0
		};
		const blinkSeed = hash * 31;
		const timing = {
			delay: blinkSeed % 40 / 10,
			duration: 2 + blinkSeed % 40 / 10
		};
		return {
			FaceComponent: FACES[faceIndex] ?? FACES[0],
			colorIndex: _colorIndex,
			rotation: position,
			blinkTimings: {
				left: timing,
				right: timing
			}
		};
	}, [
		name,
		colors?.length,
		colorClasses?.length
	]);
	const preset = INTENSITY_PRESETS[intensity3d];
	const transform = React.useMemo(() => {
		if (intensity3d === "none") return;
		return `rotateX(${isHovered && interactive ? 0 : rotation.x * preset.rotateRange}deg) rotateY(${isHovered && interactive ? 0 : rotation.y * preset.rotateRange}deg) translateZ(${preset.translateZ}px)`;
	}, [
		intensity3d,
		isHovered,
		interactive,
		rotation,
		preset
	]);
	const sizeValue = typeof size === "number" ? `${size}px` : size;
	const initial = name.charAt(0).toUpperCase();
	const bgColorClass = colorClasses?.[colorIndex];
	const bgColorHex = colors?.[colorIndex];
	const handleMouseEnter = React.useCallback((e) => {
		if (interactive) setIsHovered(true);
		onMouseEnter?.(e);
	}, [interactive, onMouseEnter]);
	const handleMouseLeave = React.useCallback((e) => {
		if (interactive) setIsHovered(false);
		onMouseLeave?.(e);
	}, [interactive, onMouseLeave]);
	return /* @__PURE__ */ jsxs("div", {
		className: [bgColorClass, className].filter(Boolean).join(" "),
		"data-facehash": "",
		"data-interactive": interactive || void 0,
		onMouseEnter: handleMouseEnter,
		onMouseLeave: handleMouseLeave,
		ref,
		style: {
			width: sizeValue,
			height: sizeValue,
			position: "relative",
			display: "flex",
			alignItems: "center",
			justifyContent: "center",
			overflow: "hidden",
			containerType: "size",
			...intensity3d !== "none" && {
				perspective: preset.perspective,
				transformStyle: "preserve-3d"
			},
			...bgColorHex && !bgColorClass && { backgroundColor: bgColorHex },
			...style
		},
		...props,
		children: [variant === "gradient" && /* @__PURE__ */ jsx("div", {
			className: gradientOverlayClass,
			"data-facehash-gradient": "",
			style: {
				position: "absolute",
				inset: 0,
				pointerEvents: "none",
				zIndex: 1,
				...gradientOverlayClass ? {} : DEFAULT_GRADIENT_STYLE
			}
		}), /* @__PURE__ */ jsxs("div", {
			"data-facehash-face": "",
			style: {
				position: "absolute",
				inset: 0,
				display: "flex",
				flexDirection: "column",
				alignItems: "center",
				justifyContent: "center",
				zIndex: 2,
				transform,
				transformStyle: intensity3d !== "none" ? "preserve-3d" : void 0,
				transition: interactive ? "transform 0.3s cubic-bezier(0.4, 0, 0.2, 1)" : void 0,
				color: "#000000"
			},
			children: [/* @__PURE__ */ jsx(FaceComponent, {
				blinkTimings,
				enableBlink,
				style: {
					width: "60%",
					height: "auto",
					maxWidth: "90%",
					maxHeight: "40%"
				}
			}), onRenderMouth ? /* @__PURE__ */ jsx("div", {
				"data-facehash-mouth": "",
				style: {
					marginTop: "8%",
					display: "flex",
					alignItems: "center",
					justifyContent: "center"
				},
				children: onRenderMouth()
			}) : showInitial && /* @__PURE__ */ jsx("span", {
				"data-facehash-initial": "",
				style: {
					marginTop: "8%",
					fontSize: "26cqw",
					lineHeight: 1,
					fontFamily: "monospace",
					fontWeight: "bold"
				},
				children: initial
			})]
		})]
	});
});
Facehash.displayName = "Facehash";

//#endregion
//#region src/avatar.tsx
const AvatarContext = React.createContext(null);
/**
* Hook to access the Avatar context.
* Throws an error if used outside of Avatar.
*/
const useAvatarContext = () => {
	const context = React.useContext(AvatarContext);
	if (!context) throw new Error("Avatar compound components must be rendered within an Avatar component");
	return context;
};
/**
* Root avatar component that provides context for image loading state.
*
* @example
* ```tsx
* <Avatar className="w-10 h-10 rounded-full overflow-hidden">
*   <AvatarImage src="/photo.jpg" alt="John" />
*   <AvatarFallback name="John Doe" />
* </Avatar>
* ```
*/
const Avatar = React.forwardRef(({ children, className, style, asChild = false, ...props }, ref) => {
	const [imageLoadingStatus, setImageLoadingStatus] = React.useState("idle");
	const contextValue = React.useMemo(() => ({
		imageLoadingStatus,
		onImageLoadingStatusChange: setImageLoadingStatus
	}), [imageLoadingStatus]);
	const Element = asChild ? React.Fragment : "span";
	const elementProps = asChild ? {} : {
		ref,
		className,
		style: {
			position: "relative",
			display: "flex",
			alignItems: "center",
			justifyContent: "center",
			flexShrink: 0,
			overflow: "hidden",
			...style
		},
		"data-avatar": "",
		"data-state": imageLoadingStatus,
		...props
	};
	return /* @__PURE__ */ jsx(AvatarContext.Provider, {
		value: contextValue,
		children: /* @__PURE__ */ jsx(Element, {
			...elementProps,
			children
		})
	});
});
Avatar.displayName = "Avatar";

//#endregion
//#region src/avatar-fallback.tsx
const WHITESPACE_REGEX = /\s+/;
/**
* Extracts initials from a name string.
*/
function getInitials(name) {
	const parts = name.trim().split(WHITESPACE_REGEX);
	if (parts.length === 0) return "";
	if (parts.length === 1) return parts[0]?.charAt(0).toUpperCase() || "";
	return ((parts[0]?.charAt(0) || "") + (parts.at(-1)?.charAt(0) || "")).toUpperCase();
}
/**
* Fallback component that displays when the image fails to load.
* Uses Facehash by default, can show initials or custom content.
*
* @example
* ```tsx
* // With Facehash (default)
* <AvatarFallback name="John Doe" />
*
* // With initials
* <AvatarFallback name="John Doe" facehash={false} />
*
* // With custom content
* <AvatarFallback>
*   <UserIcon />
* </AvatarFallback>
* ```
*/
const AvatarFallback = React.forwardRef(({ name = "", delayMs = 0, children, facehash = true, facehashProps, className, style, ...props }, ref) => {
	const { imageLoadingStatus } = useAvatarContext();
	const [canRender, setCanRender] = React.useState(delayMs === 0);
	React.useEffect(() => {
		if (delayMs > 0) {
			const timerId = window.setTimeout(() => setCanRender(true), delayMs);
			return () => window.clearTimeout(timerId);
		}
	}, [delayMs]);
	const initials = React.useMemo(() => getInitials(name), [name]);
	if (!(canRender && imageLoadingStatus !== "loaded" && imageLoadingStatus !== "loading")) return null;
	if (children) return /* @__PURE__ */ jsx("span", {
		className,
		"data-avatar-fallback": "",
		ref,
		style: {
			display: "flex",
			alignItems: "center",
			justifyContent: "center",
			width: "100%",
			height: "100%",
			...style
		},
		...props,
		children
	});
	if (facehash) return /* @__PURE__ */ jsx(Facehash, {
		className,
		"data-avatar-fallback": "",
		name,
		ref,
		size: "100%",
		...facehashProps,
		style: { ...style },
		...props
	});
	return /* @__PURE__ */ jsx("span", {
		className,
		"data-avatar-fallback": "",
		ref,
		style: {
			display: "flex",
			alignItems: "center",
			justifyContent: "center",
			width: "100%",
			height: "100%",
			...style
		},
		...props,
		children: initials
	});
});
AvatarFallback.displayName = "AvatarFallback";

//#endregion
//#region src/avatar-image.tsx
/**
* Image component that syncs its loading state with the Avatar context.
* Automatically hides when loading fails, allowing fallback to show.
*
* @example
* ```tsx
* <Avatar>
*   <AvatarImage src="/photo.jpg" alt="User" />
*   <AvatarFallback name="John Doe" />
* </Avatar>
* ```
*/
const AvatarImage = React.forwardRef(({ src, alt = "", className, style, onLoadingStatusChange, ...props }, ref) => {
	const { imageLoadingStatus, onImageLoadingStatusChange } = useAvatarContext();
	const imageRef = React.useRef(null);
	React.useImperativeHandle(ref, () => imageRef.current);
	const updateStatus = React.useCallback((status) => {
		onImageLoadingStatusChange(status);
		onLoadingStatusChange?.(status);
	}, [onImageLoadingStatusChange, onLoadingStatusChange]);
	React.useLayoutEffect(() => {
		if (!src) {
			updateStatus("error");
			return;
		}
		let isMounted = true;
		const image = new Image();
		const setStatus = (status) => {
			if (!isMounted) return;
			updateStatus(status);
		};
		setStatus("loading");
		image.onload = () => setStatus("loaded");
		image.onerror = () => setStatus("error");
		image.src = src;
		return () => {
			isMounted = false;
		};
	}, [src, updateStatus]);
	if (imageLoadingStatus !== "loaded") return null;
	return /* @__PURE__ */ jsx("img", {
		alt,
		className,
		"data-avatar-image": "",
		ref: imageRef,
		src: src ?? void 0,
		style: {
			aspectRatio: "1 / 1",
			width: "100%",
			height: "100%",
			objectFit: "cover",
			...style
		},
		...props
	});
});
AvatarImage.displayName = "AvatarImage";

//#endregion
export { Avatar, AvatarFallback, AvatarImage, CrossFace, CurvedFace, FACES, Facehash, LineFace, RoundFace, stringHash, useAvatarContext };
